src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -93,114 +87,112 @@ public ServiceBusHelper serviceBusHelper() {
     }
 
     @Scheduled(fixedDelayString = "${scheduling.task.scan.delay}")
-    public void processBlobs() throws IOException, StorageException, URISyntaxException {
-        for (CloudBlobContainer container : cloudBlobClient.listContainers()) {
+    public void processBlobs() throws IOException {
+        List<ContainerItem> containerItems = azureStorageHelper.listContainers().blockingGet().body().containerItems()
+            .stream()
+            .filter(c -> !c.name().equals("$web"))
+            .collect(toList());
+
+        log.info("Processing blobs in containers: {}", containerItems.stream()
+            .map(ContainerItem::name)
+            .collect(toList()));
+
+        for (ContainerItem container : containerItems) {
             processZipFiles(container);
         }
     }
 
-    private void processZipFiles(CloudBlobContainer container)
-        throws IOException, StorageException, URISyntaxException {
-        log.info("Processing blobs for container {}", container.getName());
+    private void processZipFiles(ContainerItem container)
+        throws IOException {
+        log.info("Processing blobs for container {}", container.name());
 
         ServiceBusHelper serviceBusHelper = serviceBusHelper();
 
         // Randomise iteration order to minimise lease acquire contention
         // For this purpose it's more efficient to have a collection that
         // implements RandomAccess (e.g. ArrayList)
         List<String> zipFilenames = new ArrayList<>();
+        ContainerURL containerURL = azureStorageHelper.getClient().createContainerURL(container.name());
+
         try {
-            container.listBlobs().forEach(
-                b -> zipFilenames.add(FilenameUtils.getName(b.getUri().toString()))
-            );
+            azureStorageHelper.listBlobsLazy(containerURL)
+                .blockingIterable()
+                .forEach(
+                    b -> zipFilenames.add(FilenameUtils.getName(b.name()))
+                );
             Collections.shuffle(zipFilenames);
             for (String zipFilename : zipFilenames) {
-                processZipFile(container, zipFilename, serviceBusHelper);
+                processZipFile(container.name(), containerURL, zipFilename, serviceBusHelper);
             }
         } finally {
-            if (serviceBusHelper != null) {
-                serviceBusHelper.close();
-            }
+//            if (serviceBusHelper != null) {
+//                serviceBusHelper.close();
+//            }
         }
     }
 
-    private void processZipFile(CloudBlobContainer container, String zipFilename, ServiceBusHelper serviceBusHelper)
-        throws IOException, StorageException, URISyntaxException {
+    private void processZipFile(String containerName, ContainerURL containerURL, String zipFilename, ServiceBusHelper serviceBusHelper)
```

Comment:
wow these methods are really long ... seems the complexity measurements would be quite high.
And the class size is rather large.
Maybe composing this in the end would be better ?